src/model/channel/message.rs

```diff
@@ -151,6 +151,63 @@ pub struct Message {
     pub poll: Option<Box<Poll>>,
 }
 
+// Custom deserialize function to deserialize components safely without knocking the whole message
+// out when new components are found but not supported.
+fn deserialize_components<'de, D>(deserializer: D) -> Result<Vec<ActionRow>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    struct MinComponent {
+        #[serde(rename = "type")]
+        kind: u8,
+    }
+
+    struct ComponentsVisitor;
+
+    impl<'de> Visitor<'de> for ComponentsVisitor {
+        type Value = Vec<ActionRow>;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            formatter.write_str("a sequence of ActionRow elements")
+        }
+
+        fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where
+            A: serde::de::SeqAccess<'de>,
+        {
+            let mut components = Vec::new();
+
+            while let Some(raw) = seq.next_element::<&serde_json::value::RawValue>()? {
+                let min_component: MinComponent =
+                    serde_json::from_str(raw.get()).map_err(serde::de::Error::custom)?;
+
+                // This is an action row, the only top level supported component in serenity at this
+                // time.
```

Comment:
Could you elaborate that we grab just the `ComponentKind` first to avoid parsing the entire component if it's unsupported?

Reply:
The comment on the line containing `MinComponent::deserialize` is good. The following comment doesn't need to reiterate that we deserialize only the type. I'd suggest writing just:

```
// Action rows are the only top level component supported in serenity at this time.
```

Reply:
Resolved.